### ISSUE TITLE
assert(true, "message") will never be evaluated

### ISSUE THREAD

**Issue by modelica-trac-importer** [modelica/Modelica#1164]
**Reported by thorade on 10 Jun 2013 11:30 UTC**
Section 8.3.7 of the Modelica Language Specification describes the assert statement:
 `assert(condition, message, level = AssertionLevel.error);`
 If the condition is `true`, message is **not** evaluated.
 If the condition is `false` different actions are taken depending on the level input.

Searching the MSL for `assert(true` yields 5 results, 4 of them in `Modelica.Fluid`. These will never be evaluated. Some are in the `else` branch of an ` if...elseif...else` construct, so if I understood the authors intention correctly, they should be evaluated.

----

Migrated-From: https://trac.modelica.org/Modelica/ticket/1164

**Comment by modelica-trac-importer** [comment]
**Comment by rfranke on 21 Jun 2013 20:29 UTC**
SVN 7c6963a2c40ba9b3778ba313961e59e9bee69300 fixes 4 occurences in Modelica.Fluid.Pipes and 2 occurences in Modelica.Fluid.Dissipation. The changes in Fluid do not effect simulation results, because the assertions are raised if an unknown modelStructure has been selected. If such a selection is possible at all, then equations are missing in the respective else branch and the translation should fail anyway.

Nevertheless, thank you for the bug report!

**Comment by modelica-trac-importer** [comment]
**Comment by dietmarw on 21 Jun 2013 22:40 UTC**
The remaining one is found in:
```
Modelica/Media/Water/package.mo
193:            assert(true, "No events for pT-model");
```

**Comment by modelica-trac-importer** [comment]
**Modified by otter on 27 Jun 2013 11:39 UTC**

**Comment by modelica-trac-importer** [comment]
**Comment by hubertus on 27 Jun 2013 16:52 UTC**
Fixed in 31c12768cec12bca631d2ebd52a50fb3ce2c10f9
